src/main/java/core/basesyntax/Square.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax;
+
+public class Square extends Figure {
+    private final double side;
+
+    public Square(double side) {
+        this.side = side;
+        setColor(new ColorSupplier().getRandomColor());
```

Comment:
Creating a new instance of ColorSupplier in the constructor of Square is not recommended. Instead, pass the color as a parameter to the constructor of the figure. This will make your Square class more testable and adhere to the principle of dependency inversion.